plan/plans/show.go

```diff
@@ -402,6 +406,50 @@ func (s *ShowPlan) fetchShowVariables(ctx context.Context) error {
 	return nil
 }
 
+func (s *ShowPlan) fetchShowStatus(ctx context.Context) error {
+	sessionVars := variable.GetSessionVars(ctx)
+	m := map[interface{}]interface{}{}
+
+	for _, v := range variable.StatusVars {
+		if s.Pattern != nil {
+			s.Pattern.Expr = expression.Value{Val: v.Name}
+		} else if s.Where != nil {
+			m[expression.ExprEvalIdentFunc] = func(name string) (interface{}, error) {
+				if strings.EqualFold(name, "Variable_name") {
+					return v.Name, nil
+				}
+
+				return nil, errors.Errorf("unknown field %s", name)
+			}
+		}
+
+		match, err := s.evalCondition(ctx, m)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !match {
+			continue
+		}
+
+		var value string
+		if !s.GlobalScope {
+			sv, ok := sessionVars.StatusVars[v.Name]
+			if ok {
+				value = sv
+			}
```

Comment:
mysql> show session Status like 'Comp%';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Compression   | OFF   |
+---------------+-------+
1 row in set (0.00 sec)

mysql> show global Status like 'Comp%';
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Compression   | OFF   |
+---------------+-------+
1 row in set (0.00 sec)

mysql>